crates/types/src/traits/block_contents.rs

```diff
@@ -47,20 +46,27 @@ pub trait BlockPayload:
     type Transaction: Transaction;
 
     /// Data created during block building which feeds into the block header
-    type Metadata;
+    type Metadata: Clone + Debug + Eq + Hash + Send + Sync;
+
+    /// Encoded payload.
+    type Encode<'a>: 'a + Iterator<Item = u8> + AsRef<[u8]> + Send;
```

Comment:
Do we need `AsRef` here? That basically forces the implementation to be a contiguous array of bytes which kind of defeats the purpose of using `Iterator`

Reply:
There was a build error before adding `AsRef`, but I'll try removing it and see if the error occurs again.

Reply:
Ahh, it's here in the transactions task:
```rust
let vid_disperse = vid.disperse(encoded_txns).unwrap();
```
`disperse` requires `encoded_txns` to be `AsRef<[u8]>`.

Reply:
Yes we should remove `AsRef` here. In the future I'll change jellyfish to take an iterator over `u8` instead of `AsRef<[u8]>` so that you can pass `encoded_txns` directly into `disperse`. In the meantime please just collect the iterator yourself into a `Vec` and then pass to `disperse`.